phantazm-neuron/src/main/java/com/github/phantazmnetwork/neuron/operation/PathOperation.java

```diff
@@ -0,0 +1,68 @@
+package com.github.phantazmnetwork.neuron.operation;
+
+import org.jetbrains.annotations.NotNull;
+import com.github.phantazmnetwork.neuron.agent.Agent;
+
+import java.util.concurrent.Semaphore;
```

Comment:
when did you need a Semaphore in an interface 🤨

Reply:
i was probably checking method signature and picked a random class to do it in :kindacringeiflikousesit: